enhancements/sig-architecture/58-addon-placement-annotation/README.md

```diff
@@ -0,0 +1,295 @@
+
+## Summary
+
+The proposal introduces some ways to help user to set up configuration for their addons.
+
+## Motivation
+
+We want to provide a way to help user to set up configuration for their addons with current addon APIs
+
+## Use cases
+
+1. As a user, I have a configuration for my all managed cluster addons, the addons share the same configuration.
+2. As a user, I have a configuration for each managed cluster addon, and every addon has its own configuration.
+
+## Proposal 1
+
+We want to introduce the addon configuration reference in the `ClusterManagementAddOn` and `ManagedClusterAddOn` APIs.
+
+### Design Details
+
+Currently, we have a `ConfigCoordinates` option in the `ClusterManagementAddOn` API. When multiple addons share the
+same configuration, the addons are able to locate their addon configuration with this option.
+
+```golang
+// ClusterManagementAddOnSpec provides information for the add-on.
+type ClusterManagementAddOnSpec struct {
+	// addOnMeta is a reference to the metadata information for the add-on.
+	// +optional
+	AddOnMeta AddOnMeta `json:"addOnMeta"`
+
+	// addOnConfiguration is a reference to configuration information for the add-on.
+	// In scenario where multiple add-ons share the same add-on CRD,
+	// multiple ClusterManagementAddOn resources need to be created and reference the same AddOnConfiguration.
+	// +optional
+	AddOnConfiguration ConfigCoordinates `json:"addOnConfiguration"`
+}
+
+// ConfigCoordinates represents the information for locating the CRD and CR that configures the add-on.
+type ConfigCoordinates struct {
+	// crdName is the name of the CRD used to configure instances of the managed add-on.
+	// This field should be configured if the add-on have a CRD that controls the configuration of the add-on.
+	// +optional
+	CRDName string `json:"crdName"`
+
+	// crName is the name of the CR used to configure instances of the managed add-on.
+	// This field should be configured if add-on CR have a consistent name across the all of the ManagedCluster instaces.
+	// +optional
+	CRName string `json:"crName"`
+
+	// lastObservedGeneration is the observed generation of the custom resource for the configuration of the addon.
+	// +optional
+	LastObservedGeneration int64 `json:"lastObservedGeneration,omitempty"`
+}
+```
+
+but the `ConfigCoordinates` has some limits, it only support to use `CustomResourceDefinition` as the addon
+configuration, and for addon developers, it's not easy to locate the configuration object in their controllers. we want
+to refactor the `ConfigCoordinates` to
+
+``` golang
+// ConfigCoordinates contains enough information to let one addon identify its configuration object.
+// A config object must be in the same namespace with its addon, or be cluster-scoped, so there is no
+// namespace field.
+type ConfigCoordinates struct {
+	// API version of the referent.
+	APIVersion string `json:"apiVersion"`
```

Comment:
let's use gvr here. and also default namespace/name??

Reply:
use the gvr and add the default ns and name